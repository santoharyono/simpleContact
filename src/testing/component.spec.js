import React from 'react';
import renderer from 'react-test-renderer';
import {
  CustomButton,
  CustomTextInput,
  RoundButton,
} from '../components/molecules';

describe('Render customButton', () => {
  it('render customButton correctly', () => {
    const CustomButtonComponent = renderer.create(<CustomButton />).toJSON();
    expect(CustomButtonComponent).toMatchSnapshot();
  });
});

describe('Render customTextInput', () => {
  it('render customTextInput correctly', () => {
    const CustomTextInputComponent = renderer
      .create(<CustomTextInput />)
      .toJSON();
    expect(CustomTextInputComponent).toMatchSnapshot();
  });
});

describe('Render roundButton', () => {
  it('render roundButton correctly', () => {
    const RoundButtonComponent = renderer.create(<RoundButton />).toJSON();
    expect(RoundButtonComponent).toMatchSnapshot();
  });
});
