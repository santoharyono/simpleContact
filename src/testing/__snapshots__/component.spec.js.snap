// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render customButton render customButton correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;

exports[`Render customTextInput render customTextInput correctly 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#c7c7c7",
      "borderRadius": 10,
      "marginVertical": "5%",
      "paddingHorizontal": "5%",
      "width": "100%",
    }
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`Render roundButton render roundButton correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 50,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "width": 70,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    +
  </Text>
</View>
`;

exports[`render customButton correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
